src/global/interfaces/link-button-properties.ts

```diff
@@ -126,56 +115,88 @@ export function getLinkButtonBaseAttributeList(
 }
 
 /**
- * Creates a list of attributes for the <a> or <button> tags, depending on whether the href attribute value is present.
- * @param elementId HTML element's id
- * @param elementClassName HTML element's CSS class
+ * Creates the basic attribute list for the link/button tag; undefined/null properties are not set.
+ * @param accessibilityProps accessibility props
+ */
+export function getLinkButtonBaseAttributeList(
+  accessibilityProps: AccessibilityProperties
+): Record<string, string> {
+  return Object.assign(
+    { dir: getDocumentWritingMode() },
+    getAccessibilityAttributeList(accessibilityProps)
+  );
+}
+
+/**
+ * Lists all attributes for a link; undefined/null properties are not set.
+ * @param linkProperties link properties
+ * @param buttonProperties (optional) In the case of a mixed button and anchor variant, pass in also button properties, which enables the possibility of a disabled link
+ */
+export function getLinkAttributeList(
+  linkProperties: LinkProperties,
+  buttonProperties?: ButtonProperties
+): Record<string, string> {
+  const baseAttributeList = getLinkButtonBaseAttributeList(linkProperties);
+
+  if (!linkProperties.href) {
+    return baseAttributeList;
+  }
+
+  const linkAttributeList = Object.assign(baseAttributeList, {
+    href: linkProperties.href,
+    ...(linkProperties.download && { download: '' }),
+    ...(buttonProperties.disabled && { tabIndex: '-1' }),
```

Comment:
This pattern is not necessary. Object.assign filters out undefined values.
```suggestion
    download: linkProperties.download ? '' : undefined,
    tabIndex: buttonProperties.disabled ? '-1' : undefined,
```